### PR TITLE
Eloquent js chapter4

### PR DESCRIPTION
**Pull request by DGEs2018**
Could you please review the files reversing-an-array.js & the-sum-of-a-range.js files "regardless the wip commit message" from chapter 4 ? Thanks in advance.

### CODE REVIEW

File: Week9/eloquent-js-chapter4-exercises/a-list.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Write a function that builds list: with two properties, each of which containing nested propertiees 
+ */
+function arrayToList(array) {
+	let arrayHolder = null;
+	for (let i = array.length - 1; i >= 0; i--) {
+		list = { value1: array[i], rest: list };
```
Comment (sophialittlejohn) [inline Week9/eloquent-js-chapter4-exercises/a-list.js]: the variable `list` has appeared out of the blue 🤔

File: Week9/eloquent-js-chapter4-exercises/a-list.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+console.assert(arrayToList([ 10, 20 ]), "This ain't working yet!");
+// → {value: 10, rest: {value: 20, rest: null}}
+/* console.log(listToArray(arrayToList([ 10, 20, 30 ])));
+// → [10, 20, 30]
+console.log(prepend(10, prepend(20, null)));
+// → {value: 10, rest: {value: 20, rest: null}}
+console.log(nth(arrayToList([ 10, 20, 30 ]), 1));
+// → 2
+ */
```
Comment (sophialittlejohn) [inline Week9/eloquent-js-chapter4-exercises/a-list.js]: Please only commit files the run

File: Week8/journal.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        Object.assign(objectExample, {a: 4, b:5, c:6}) // returns {x: 1, y: 2, z: 3, a: 4, b: 5, c: 6}
+        ```
+    - Mutability
+        - The value of numbers, strings and Boolean datatypes can not be changed (immutable)
+        - On the contrary objects allow change
+       ``` let myObject = {value: 'don't know what to give it'}
+        let theirObject = myObject; // Objects are said to have the same identity
+        ```
+        - myTodos(['wake up', 'drink a coffee', 'get the train', 'do the commute', 'be at work']);
+        - Delete branch - read-up resources
```
Comment (sophialittlejohn) [inline Week8/journal.md]: Also this doens't belong here 😉

File: Week8/journal.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        
+        ```let myPersonalDetails = {
+            firstname: "Dawit",
+            lastname: "Ghebremedhin"
+            Addresse : "Wagnerstrasse 21"  
+        }
+        Object.assign - function that copies all properties from one to another object
+
+        let objectExample = {x:1, y:2, z:3}
+        Object.assign(objectExample, {a: 4, b:5, c:6}) // returns {x: 1, y: 2, z: 3, a: 4, b: 5, c: 6}
```
[the remark this answers is not in the document]
Comment (DGEs2018) [reply Week8/journal.md]: Both .slice() & .slice(0) do it.
let exArray = ['Just', 'a', 'demo', 'to', 'show', 'how', 'slice', 'works'];
Running exArray.slice(0);, exArray.slice(0); should return the exact copy

[the remark this answers is not in the document]
Comment (sophialittlejohn) [reply Week8/journal.md]: 🤓👏

File: Week9/eloquent-js-chapter4-exercises/a-list.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    return listContainer;
+}
+
+function prepend(value, list) {
+	return { value, rest: list };
+}
+
+function nth(list, n) {
+	if (!list) return undefined;
+	else if (n == 0) return list.value;
```
Comment (yhabib) [inline Week9/eloquent-js-chapter4-exercises/a-list.js]: Missing last condition where you make the recursive call

File: Week9/eloquent-js-chapter4-exercises/a-list.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	/* const arrStorer = [];
+	for (let item of array) {
+		arrStorer.unshift(item);
+	}
+	return arrStorer;
+} */
+
+function listToArray(list) {
+	let listContainer = [];
+	for (let node = list; node; node.test) {
```
Comment (yhabib) [inline Week9/eloquent-js-chapter4-exercises/a-list.js]: Typo `node.test` should be `node.rest`.

**Pull request merged**